### PR TITLE
[FLINK-20168][docs-zh] Translate page 'Flink Architecture' into Chinese.

### PR DESCRIPTION
**Pull request by caozhen1937**
## What is the purpose of the change

Translate page 'Flink Architecture' into Chinese.

## Brief change log

*docs/concepts/flink-architecture.zh.md*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive):(no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by caozhen1937** [comment]
hi @wuchong ,please help to review if you have time.

**Review by HuangXingBo — CHANGES_REQUESTED** [review]
Hi, you need to rebase `master` branch code first

**Comment by caozhen1937** [comment]
@HuangXingBo  thank you, I have rebase `master`.

File: docs/concepts/flink-architecture.zh.md [hunk 1]
```diff
@@ -24,229 +24,109 @@ specific language governing permissions and limitations
...
-## Tasks and Operator Chains
+## Tasks 和算子链
 
-For distributed execution, Flink *chains* operator subtasks together into
-*tasks*. Each task is executed by one thread.  Chaining operators together into
-tasks is a useful optimization: it reduces the overhead of thread-to-thread
-handover and buffering, and increases overall throughput while decreasing
-latency.  The chaining behavior can be configured; see the [chaining docs]({%
-link dev/stream/operators/index.zh.md %}#task-chaining-and-resource-groups) for details.
+对于分布式执行，Flink 将算子的 subtasks *链接*成 *tasks*。每个 task 由一个线程执行。将算子链接成 task 是个有用的优化：它减少线程间切换、缓冲的消耗，并且减少延迟的同时增加整体吞吐量。链行为是可以配置的；请参考[链文档]({% link dev/stream/operators/index.zh.md %}#task-chaining-and-resource-groups)以获取详细信息。
```
Comment (HuangXingBo) [inline docs/concepts/flink-architecture.zh.md:72]:
```suggestion
对于分布式执行，Flink 将算子的 subtasks *链接*成 *tasks*。每个 task 由一个线程执行。将算子链接成 task 是个有用的优化：它减少线程间切换和缓冲的开销，并且减少延迟的同时增加整体吞吐量。链行为是可以配置的；请参考[链文档]({% link dev/stream/operators/index.zh.md %}#task-chaining-and-resource-groups)以获取详细信息。
```

**Review by HuangXingBo — COMMENTED** [review]
@caozhen1937 Thanks a lot for the contribution. I have left some comments.

File: docs/concepts/flink-architecture.zh.md [hunk 2]
```diff
@@ -24,229 +24,109 @@ specific language governing permissions and limitations
...
+本节概述了 Flink 架构，并且描述了其主要组件如何交互以执行应用程序和从故障中恢复。
 
 * This will be replaced by the TOC
 {:toc}
 
-## Anatomy of a Flink Cluster
+## Flink 集群剖析
 
-The Flink runtime consists of two types of processes: a _JobManager_ and one or more _TaskManagers_.
+Flink 运行时由两种类型的进程组成：_JobManager_和一个或者多个_TaskManager_。
```
[the remark this answers is not in the document]
Comment (caozhen1937) [reply docs/concepts/flink-architecture.zh.md:36]: 这里说的是两种类型的进程，我觉得只是说明进程名称就可以，不需要指明个数，并且JobManager可能不是一个。

File: docs/concepts/flink-architecture.zh.md [hunk 3]
```diff
@@ -24,229 +24,109 @@ specific language governing permissions and limitations
...
 
-The *TaskManagers* (also called *workers*) execute the tasks of a dataflow, and buffer and exchange the data
-streams.
+*TaskManager*（也称为 *worker*）执行数据流的 task，并且缓存和交换数据流。
 
-There must always be at least one TaskManager. The smallest unit of resource scheduling in a TaskManager is a task _slot_. The number of task slots in a
-TaskManager indicates the number of concurrent processing tasks. Note that
-multiple operators may execute in a task slot (see [Tasks and Operator
-Chains](#tasks-and-operator-chains)).
+必须始终至少有一个 TaskManager。在 TaskManager 中资源调度的最小单位是 task _slot_。TaskManager 中 task slot 的数量表示并发处理 task 的数量。请注意一个 task slot 中可以执行多个算子（请参考[Tasks 和算子链](#tasks-and-operator-chains)）。
```
[the remark this answers is not in the document]
Comment (caozhen1937) [reply docs/concepts/flink-architecture.zh.md:66]: https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications

File: docs/concepts/flink-architecture.zh.md [hunk 4]
```diff
@@ -24,229 +24,109 @@ specific language governing permissions and limitations
...
 
 {% top %}
 
-## Tasks and Operator Chains
+## Tasks 和算子链
```
[the remark this answers is not in the document]
Comment (caozhen1937) [reply docs/concepts/flink-architecture.zh.md:70]: ditto.

File: docs/concepts/flink-architecture.zh.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (HuangXingBo) [reply docs/concepts/flink-architecture.zh.md:66]: Ok. make sense

File: docs/concepts/flink-architecture.zh.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (HuangXingBo) [reply docs/concepts/flink-architecture.zh.md:36]: 我觉得这里的描述很可能是针对下面那个图来的。你TaskManager在聊数量了，前面JobManager又不聊，会让读者很奇怪吧，而且也许和英文的a JobManager保持一致也许会更好。你怎么看？

File: docs/concepts/flink-architecture.zh.md [hunk 5]
```diff
@@ -24,229 +24,109 @@ specific language governing permissions and limitations
...
-The JobManager and TaskManagers can be started in various ways: directly on
-the machines as a [standalone cluster]({% link
-deployment/resource-providers/standalone/index.zh.md %}), in containers, or managed by resource
-frameworks like [YARN]({% link deployment/resource-providers/yarn.zh.md
-%}) or [Mesos]({% link deployment/resource-providers/mesos.zh.md %}).
-TaskManagers connect to JobManagers, announcing themselves as available, and
-are assigned work.
+*Client* 不是运行时和程序执行的一部分，而是用于准备数据流并将其发送给 JobManager。之后，客户端可以断开连接（_分离模式_），或保持连接来接收进程报告（_附加模式_）。客户端可以作为触发执行 Java/Scala 程序的一部分运行，也可以在命令行进程`./bin/flink run ...`中运行。
+
+可以通过各种方式启动 JobManager 和 TaskManager：直接在机器上作为[standalone 集群]({% link deployment/resource-providers/standalone/index.zh.md %})启动、在容器中启动、或者通过[YARN]({% link deployment/resource-providers/yarn.zh.md %})或[Mesos]({% link deployment/resource-providers/mesos.zh.md %})等资源框架管理启动。TaskManager 连接到 JobManagers，宣布自己可用，并被分配工作。
```
[the remark this answers is not in the document]
Comment (caozhen1937) [reply docs/concepts/flink-architecture.zh.md:42]: 这里的 managed 是不是想表达使用资源框架 “管理并启动”？

File: docs/concepts/flink-architecture.zh.md [hunk 6]
```diff
@@ -24,229 +24,109 @@ specific language governing permissions and limitations
...
-There is always at least one JobManager. A high-availability setup might have
-multiple JobManagers, one of which is always the *leader*, and the others are
-*standby* (see [High Availability (HA)]({% link deployment/ha/index.zh.md %})).
+始终至少有一个 JobManager。高可用设置中可能有多个 JobManager，其中一个始终是 *leader*，其他的则是 *standby*（请参考 [高可用（HA）]({% link deployment/ha/index.zh.md %})）。
 
 ### TaskManagers
 
-The *TaskManagers* (also called *workers*) execute the tasks of a dataflow, and buffer and exchange the data
-streams.
+*TaskManager*（也称为 *worker*）执行数据流的 task，并且缓存和交换数据流。
```
[the remark this answers is not in the document]
Comment (caozhen1937) [reply docs/concepts/flink-architecture.zh.md:64]: 我觉得这里的 `一个 dataflow`有点直译了，`dataflow` 翻译为数据流的话和后面的数据流有点重复，`dataflow`可不可以翻译为`作业流`？我想这里应该是想表达作业的意思。

File: docs/concepts/flink-architecture.zh.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (caozhen1937) [reply docs/concepts/flink-architecture.zh.md:36]: 同意，如果是针对下面那个图，确实应该翻译成一个。

File: docs/concepts/flink-architecture.zh.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (HuangXingBo) [reply docs/concepts/flink-architecture.zh.md]: 是的，你这样翻译更好

File: docs/concepts/flink-architecture.zh.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (HuangXingBo) [reply docs/concepts/flink-architecture.zh.md]: 是的，翻译成`数据流`的话，就会让读者和后面的`data stream`混在一起理解，造成误解。`作业流`的话，我也不知道主流翻译有没有这样翻译的。

**Review by HuangXingBo — APPROVED** [review]
@caozhen1937 Thanks a lot for the update. LGTM.
@wuchong Could you help review it? Thanks.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request closed without merging**